From: Oliver Neukum <oneukum@suse.de>
Date: Fri, 11 Feb 2011 12:35:05 +0100
Subject: [PATCH] btusb:fix crash with quirky dongles doing sound
References: bnc#671256
Patch-mainline: submitted

Quirky dongles sometimes do not use the iso interface which
causes a crash with runtime PM

Signed-off-by: Oliver Neukum <oneukum@suse.de>
---
--- linux-2.6.32-SLE11-SP1/drivers/bluetooth/btusb.c.alt	2011-02-11 13:31:33.502148000 +0100
+++ linux-2.6.32-SLE11-SP1/drivers/bluetooth/btusb.c	2011-02-11 13:33:08.112689000 +0100
@@ -821,7 +821,7 @@ static void btusb_work(struct work_struc
 
 	if (hdev->conn_hash.sco_num > 0) {
 		if (!data->did_iso_resume) {
-			err = usb_autopm_get_interface(data->isoc);
+			err = usb_autopm_get_interface(data->isoc ? data->isoc : data->intf);
 			if (err < 0) {
 				clear_bit(BTUSB_ISOC_RUNNING, &data->flags);
 				usb_kill_anchored_urbs(&data->isoc_anchor);
@@ -851,7 +851,7 @@ static void btusb_work(struct work_struc
 		__set_isoc_interface(hdev, 0);
 		if (data->did_iso_resume) {
 			data->did_iso_resume = 0;
-			usb_autopm_put_interface(data->isoc);
+			usb_autopm_put_interface(data->isoc ? data->isoc : data->intf);
 		}
 	}
 }
